internal/featuretests/v3/extensionservice_test.go

```diff
@@ -402,6 +404,192 @@ func extInvalidLoadBalancerPolicy(t *testing.T, rh ResourceEventHandlerWrapper,
+		Spec: contour_v1alpha1.ExtensionServiceSpec{
+			Services: []contour_v1alpha1.ExtensionServiceTarget{
+				{Name: "svc1", Port: 8081},
+				{Name: "svc2", Port: 8082},
+			},
+			LoadBalancerPolicy: &contour_v1.LoadBalancerPolicy{
+				Strategy: "Cookie",
+			},
+			CircuitBreakerPolicy: &contour_v1alpha1.CircuitBreaker{
+				MaxConnections:        10000,
+				MaxPendingRequests:    1048,
+				MaxRequests:           494,
+				MaxRetries:            10,
+				PerHostMaxConnections: 1,
+			},
+		},
+	}
+
+	rh.OnAdd(ext)
+
+	c.Request(clusterType).Equals(&envoy_service_discovery_v3.DiscoveryResponse{
+		TypeUrl: clusterType,
+		Resources: resources(t,
+			DefaultCluster(
+				// Default load balancer policy should be set as we were passed
+				// an invalid value, we can assert we get a basic cluster.
+				h2cCluster(cluster("extension/ns/ext", "extension/ns/ext", "extension_ns_ext")),
+				&envoy_config_cluster_v3.Cluster{
+					TransportSocket: envoy_v3.UpstreamTLSTransportSocket(
+						&envoy_transport_socket_tls_v3.UpstreamTlsContext{
+							CommonTlsContext: &envoy_transport_socket_tls_v3.CommonTlsContext{
+								AlpnProtocols: []string{"h2"},
+							},
+						},
+					),
+					CircuitBreakers: &envoy_config_cluster_v3.CircuitBreakers{
+						Thresholds: []*envoy_config_cluster_v3.CircuitBreakers_Thresholds{{
+							MaxConnections:     wrapperspb.UInt32(10000),
+							MaxPendingRequests: wrapperspb.UInt32(1048),
+							MaxRequests:        wrapperspb.UInt32(494),
+							MaxRetries:         wrapperspb.UInt32(10),
+							TrackRemaining:     true,
+						}},
+						PerHostThresholds: []*envoy_config_cluster_v3.CircuitBreakers_Thresholds{{
+							MaxConnections: wrapperspb.UInt32(1),
+							TrackRemaining: true,
+						}},
+					},
+				},
+			),
+		),
+	})
+
+	rh.OnUpdate(ext, &contour_v1alpha1.ExtensionService{
+		ObjectMeta: fixture.ObjectMeta("ns/ext"),
+		Spec: contour_v1alpha1.ExtensionServiceSpec{
+			Services: []contour_v1alpha1.ExtensionServiceTarget{
+				{Name: "svc1", Port: 8081},
+				{Name: "svc2", Port: 8082},
+			},
+			LoadBalancerPolicy: &contour_v1.LoadBalancerPolicy{
+				Strategy: "RequestHash",
+			},
+		},
+	})
+
+	c.Request(clusterType).Equals(&envoy_service_discovery_v3.DiscoveryResponse{
+		TypeUrl: clusterType,
+		Resources: resources(t,
+			DefaultCluster(
+				// Default load balancer policy should be set as we were passed
+				// an invalid value, we can assert we get a basic cluster.
+				h2cCluster(cluster("extension/ns/ext", "extension/ns/ext", "extension_ns_ext")),
+				&envoy_config_cluster_v3.Cluster{
+					TransportSocket: envoy_v3.UpstreamTLSTransportSocket(
+						&envoy_transport_socket_tls_v3.UpstreamTlsContext{
+							CommonTlsContext: &envoy_transport_socket_tls_v3.CommonTlsContext{
+								AlpnProtocols: []string{"h2"},
+							},
+						},
+					),
+				},
+			),
+		),
+	})
+}
+
+func extGlobalCircuitBreakers(t *testing.T, rh ResourceEventHandlerWrapper, c *Contour) {
```

Comment:
we could add one more test, ensuring global circuit breaker settings are overridden by those set on an ExtensionService, per the logic added to the extension processor

```go
	if p.GlobalCircuitBreakerDefaults != nil {
	...
	}

	if ext.Spec.CircuitBreakerPolicy != nil {
	...

Reply:
added in https://github.com/projectcontour/contour/pull/6539/commits/4790f07b4d94965415cb8ef2ac8004949da334e1